scripts/simplify_network.py

```diff
@@ -620,6 +679,8 @@ def consense(x):
 
     update_p_nom_max(n)
 
+    merge_isolated_buses(n)
```

Comment:
Recalling above, the proposal is to decompose the problem into two pieces: 1. dropping isolated nodes below a threashold, 2 merge isolated nodes below another threashold.
Note: when the value of a threshold is False, then that function is not abilited, it is true, that function is applied to all isolated nodes.

Reply:
Agree, it sounds absolutely reasonable. A question: can we assume that if all isolated buses are dropped, the whole workflow should work with the grouping by country and sub-network? I'd suppose yes as it's the limit case of the final approach

Reply:
Obviously the determine_topology function should be repeated after each cleaning procedure to update the values of sub_network.
If we drop all isolated buses, the cluster_network procedure would still work with ["countries", "sub_network"] but it will be equivalent to "countries" as long as all country is completely interconnected.

Even when all isolated buses are dropped the clustering shall be done by "countries" and "sub_network"

Reply:
Here I'd expect something like:

``` python

e_threshold_drop_isolated = config["simplify_network"].get("e_threshold_drop_isolated ", False)
if e_threshold_drop_isolated :
   drop_isolated_nodes(n, e_threshold_isolated_buses)

e_threshold_mesge_isolated = config["simplify_network"].get("e_threshold_mesge_isolated ", False)
if e_threshold_mesge_isolated:
   merge_isolated_buses(n, e_threshold_mesge_isolated)
```

naming of options may be improved and the config management adjusted; that's a concept.
Ideally we can start from the first function and see if the issue was due to the second block, which I think is better handled with the busmap clustering

Reply:
Thank you very much for the explanations!
Was not aware on `determine_topology` usage which seems to be very relevant

Reply:
A clarification question: do you think it makes sense to have two different thresholds? Would it be alright to leave isolated buses which have load between the thresholds? Just trying to understand the concept you are suggesting :)

Reply:
Absolutely: 2 thresholds, one to drop and the other one to merge.
I'd expect the merge to be larger than dropping but no check is needed for this: if the converse applies, no buses are merged as all the relevant isolated are dropped.

Reply:
Thank you!
A question: wouldn't it be more convenient to specify the thresholds as a share of the annual load? An absolute value might be not always obvious in advance...

Reply:
I'd say let's keep it simple for now.
Otherwise mathematically it becomes more complex. It is ok with a numeric threshold

Reply:
Ok!